Extensions/SplashScreen/SplashScreen.ps1

```diff
@@ -661,7 +670,12 @@ function OnGameStarting
         $logoPath = ""
         if ($settings.showLogoInSplashscreen -eq $true)
         {
-            $logoPath = [System.IO.Path]::Combine($PlayniteApi.Paths.ConfigurationPath, "ExtraMetadata", "games", $game.Id, "Logo.png")
+            if ($settings.useIconAsLogo -eq $true)
+            {
+                $logoPath = $PlayniteApi.Database.GetFullFilePath($game.Icon)
```

Comment:
I moved the logo search logic in a new method in the new commit.  There I implemented the null/url check. 
The .ico-format is already displayed correctly. You could only argue that it doesn´t make much sense to show an icon because it´s resolution is certainly to small to be displayed this big. Maybe there should be some sort of acceptable file extension for the icon to be displayed. Depending on how far we want to go, we could make that choice user configurable.